### ISSUE TITLE
Npm install directory detected as DL3016

### ISSUE THREAD

**Issue by SuperSandro2000** [hadolint/hadolint#356]
- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior

This shouldn't be DL3016 as it installs a package.json in a directory.
```
RUN npm install /app
```

### Actual behavior

```
amd64.Dockerfile:29 DL3016 Pin versions in npm. Instead of `npm install <package>` use `npm install <package>@<version>`
```

### Steps to reproduce the behavior

hadolint Dockerfile

**Output of `hadolint --version` or
  `docker run --rm hadolint/hadolint hadolint --version`:**

```bash
Haskell Dockerfile Linter v1.17.2-0-g574b169

```

**Dockerfile** (if relevant)

```Dockerfile
FROM node:alpine

RUN apk add --no-cache --no-progress chromium git \
  && git clone https://github.com/prerender/prerender.git /app \
  && apk del git

RUN npm install /app
```

**Additional environment details (OS, stack version, etc.)**

**Comment by simonweil** [comment]
I'm getting the same lint error for 

`RUN npm install dompurify@1.0.10 jsdom@15.1.1 --prefix /home/ubuntu --no-package-lock`
